backend/src/akvo/flow_services/uploader.clj

```diff
@@ -262,6 +262,18 @@
       (.endsWith uname "XLSX") (raw-data (io/file path filename) base-url bucket-name surveyId) ; Upload raw data
       :else (upload (io/file path) bucket-name))))
 
+(defn- add-image-upload-messages
+  [app-id file-name form-instance-id responses]
+  (let [bucket (:s3bucket (config/find-config app-id))
+        total-images (count responses)
+        grouped-responses (group-by :status responses)
+        success-msg (partial (format "File: '%1$s' processed. '%4$s' out of '%3$s' images successfully added to instance id: '%2$s'") file-name form-instance-id total-images)
+        failure-msg (partial (format "Failed to add '%4$s' out of '%2$s' image files for folder '%3$s' in file '%1$s': %5$s ") file-name total-images form-instance-id)]
+    (doseq [response grouped-responses]
+      (if (= (:status response) 200)
```

Comment:
```clojure
(doseq [[k v :as x] {1 ["a"], 2 ["as" "aa"], 3 ["asd"], 4 ["asdf" "qwer"]}]
  (println k v x))

```
output
```clojure
1 [a] [1 [a]]
2 [as aa] [2 [as aa]]
3 [asd] [3 [asd]]
4 [asdf qwer] [4 [asdf qwer]]
```
If you use doseq with a [group-by](https://clojuredocs.org/clojure.core/group-by) structure you don't have keys available (as `:status`  in each iteration)

Also using [doseq](https://clojuredocs.org/clojure.core/doseq) returns nil 

proposing in a new commit mapv instead (that is also eager)